src/interlocking_component/route_controller.py

```diff
@@ -307,6 +307,23 @@ def maybe_set_fahrstrasse(self, train: Train, edge: Edge):
             print("No new route needed")
             return
 
+        try:
+            assert train.edge.track.reservations[0][1] == train.edge.identifier
+            assert train.edge.track == train.reserved_tracks[0]
+            if len(train.reserved_tracks) > 1:
+                assert (
+                    train.edge.identifier
+                    == train.reserved_tracks[1].reservations[0][1].identifier
+                )
+        except Exception as exc:
+            for track in train.reserved_tracks:
+                print(track.edges[0].identifier)
+            raise ValueError(
+                ("Something is off: Track segment " f"{train.edge.identifier}")
+            ) from exc
+        train.edge.track.reservations.pop(0)
+        train.reserved_tracks.pop(0)
```

Comment:
Why is this in this PR? By now is is handled in the infrastructure provider which calls the RouteController.